cmd/sonrd/utils/keychain.go

```diff
@@ -0,0 +1,123 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/gob"
+
+	"github.com/pkg/errors"
+	"github.com/sonr-io/keyring"
+	"github.com/sonr-io/sonr/pkg/crypto/mpc"
+	"github.com/sonr-io/sonr/pkg/fs"
+	mt "github.com/sonr-io/sonr/third_party/types/motor"
+)
+
+const K_SERVICE_NAME = "sonr-dev"
+
+type UserAuth struct {
+	Password  string
+	AesDSCKey []byte
+	AesPSKKey []byte
+}
+
+func (i UserAuth) Validate() bool {
+	if len(i.Password) < 12 {
+		return false
+	}
+	if i.AesDSCKey == nil {
+		return false
+	}
+	if len(i.AesDSCKey) < 32 {
```

Comment:
This is weird.

AES Keys can only be 128, 192, or 256 bytes, meaning this key has to be 32 bytes or 256 bits. Therefore, should this check not be strict equality?

Reply:
@mcjcloud ?